internal/plugin/grpc_client_plugin.go

```diff
@@ -161,13 +172,26 @@ func (gc *GrpcClient) Process(ctx context.Context, msg *bus.Message) {
 	switch msg.Topic {
 	case bus.InstancesTopic:
 		if newInstances, ok := msg.Data.([]*v1.Instance); ok {
+			gc.instancesMutex.Lock()
+			gc.instances = newInstances
+			gc.instancesMutex.Unlock()
+
 			err := gc.sendDataPlaneStatusUpdate(ctx, newInstances)
 			if err != nil {
 				slog.ErrorContext(ctx, "Unable to send data plane status update", "error", err)
 			}
 		}
 	case bus.GrpcConnectedTopic:
 		slog.DebugContext(ctx, "Agent connected")
+		gc.isConnected.Store(true)
+
+		gc.instancesMutex.Lock()
+		defer gc.instancesMutex.Unlock()
```

Comment:
Small side note, while this is not a bug right now, this has the potential to introduce bugs if we add code after the `switch`, as the `defer` is only executed at the end of **function** block, not the current code block (i.e. the `case` block of this `switch`). So it would technically be safer to manually unlock it after line 191.

This could be something to mention in our style guide, assuming it is not already in the Uber one.

Reply:
thats a fair point. Let me update that

Reply:
updated